orderer/consensus/follower/follower_chain.go

```diff
@@ -257,14 +331,92 @@ func (c *Chain) pull() error {
 // pullUpToJoin pulls blocks up to the join-block height without inspecting membership on fetched config blocks.
 // It checks whether the chain was stopped between blocks.
 func (c *Chain) pullUpToJoin() error {
-	//TODO
-	return errors.New("not implemented yet: pull up to join block")
+	targetHeight := c.joinBlock.Header.Number + 1
+	firstBlockToPull := c.support.Height()
+	c.logger.Debugf("first block to pull: %d, target height: %d", firstBlockToPull, targetHeight)
+	if firstBlockToPull >= targetHeight {
+		c.logger.Infof("Target height (%d) is <= to our ledger height (%d), skipping pulling", targetHeight, firstBlockToPull)
+		return nil
+	}
+
+	puller, err := c.createBlockPullerFunc()
+	if err != nil {
+		c.logger.Errorf("Error creating block puller: %s", err)
+		return errors.Wrap(err, "error creating block puller")
+	}
+	defer puller.Close()
+
+	var nextBlock *common.Block
+	var actualPrevHash []byte
+
+	// Pull the first block
```

Comment:
I don't get it, what's so special about the 1st block? why can't we pull it along the rest below?

Reply:
We need to compare the prevHash of the first block to the header hash of the previous block from the ledger, that is all.

Reply:
Why? it seems to me you are pulling and checking your ledger is consistent with what you're going to pull next, i don't understand why you're doing this for the 1st block?

Reply:
I think I understand what you mean. Basically I want to initialize the actualPrevHash before the main loop. So I only nedd to either:
- pull the last block from the ledger and compute its hash, or
- pull the genesis block, compute its hash, and append it.
There is no need to pull the "next" block as I do there. I will correct that.

Reply:
Done